File: sample_code/java/CoverageVerification.java
package com.testable.demo;

/** Whitebox verification harness — exercises decision paths and logical combinations. */
public final class CoverageVerification {

    private CoverageVerification() {}

    public static void verifyRouteRequestPaths() {
        assert "405".equals(new PaymentRouter().route("DELETE", true, false));
        assert "405".equals(new PaymentRouter().route("PATCH", false, false));

        assert "200-read".equals(new PaymentRouter().route("GET", true, false));
        assert "201-created".equals(new PaymentRouter().route("POST", true, false));
        assert "202-accepted".equals(new PaymentRouter().route("PUT", true, false));

        assert "401-unauthorized".equals(new PaymentRouter().route("GET", false, false));
        assert "401-unauthorized".equals(new PaymentRouter().route("GET", true, true));
        assert "401-unauthorized".equals(new PaymentRouter().route("POST", false, true));
    }

    public static void verifyEvaluateFlagsCombinations() {
        PaymentRouter router = new PaymentRouter();
        assert "path-alpha".equals(router.evaluateFlags(true, true, false));
        assert "path-alpha".equals(router.evaluateFlags(true, true, true));
        assert "path-alpha".equals(router.evaluateFlags(true, false, true));
        assert "path-alpha".equals(router.evaluateFlags(true, false, false));

        assert "path-beta".equals(router.evaluateFlags(false, true, false));
        assert "path-beta".equals(router.evaluateFlags(false, true, true));
        assert "path-beta".equals(router.evaluateFlags(false, false, true));

        assert "path-default".equals(router.evaluateFlags(false, false, false));
    }

    public static void verifyCompoundConditions() {
        PaymentRouter router = new PaymentRouter();
        assert "401-unauthorized".equals(router.route("GET", true, !false == false));
        assert "200-read".equals(router.route("GET", true, !true == false));
        assert "path-alpha".equals(router.evaluateFlags(!false, !false, !true));
        assert "path-beta".equals(router.evaluateFlags(!true, !false, !false));
    }

    public static void runAllVerifications() {
        verifyRouteRequestPaths();
        verifyEvaluateFlagsCombinations();
        verifyCompoundConditions();
    }
}

File: sample_code/csharp/CoverageVerification.cs
using System.Diagnostics;

namespace Testable.Demo;

/// <summary>Whitebox verification harness — exercises decision paths and logical combinations.</summary>
public static class CoverageVerification
{
    public static void VerifyRouteRequestPaths()
    {
        Debug.Assert(PaymentRouter.Route("DELETE", true, false) == "405");
        Debug.Assert(PaymentRouter.Route("PATCH", false, false) == "405");

        Debug.Assert(PaymentRouter.Route("GET", true, false) == "200-read");
        Debug.Assert(PaymentRouter.Route("POST", true, false) == "201-created");
        Debug.Assert(PaymentRouter.Route("PUT", true, false) == "202-accepted");

        Debug.Assert(PaymentRouter.Route("GET", false, false) == "401-unauthorized");
        Debug.Assert(PaymentRouter.Route("GET", true, true) == "401-unauthorized");
        Debug.Assert(PaymentRouter.Route("POST", false, true) == "401-unauthorized");
    }

    public static void VerifyEvaluateFlagsCombinations()
    {
        Debug.Assert(PaymentRouter.EvaluateFlags(true, true, false) == "path-alpha");
        Debug.Assert(PaymentRouter.EvaluateFlags(true, true, true) == "path-alpha");
        Debug.Assert(PaymentRouter.EvaluateFlags(true, false, true) == "path-alpha");
        Debug.Assert(PaymentRouter.EvaluateFlags(true, false, false) == "path-alpha");

        Debug.Assert(PaymentRouter.EvaluateFlags(false, true, false) == "path-beta");
        Debug.Assert(PaymentRouter.EvaluateFlags(false, true, true) == "path-beta");
        Debug.Assert(PaymentRouter.EvaluateFlags(false, false, true) == "path-beta");

        Debug.Assert(PaymentRouter.EvaluateFlags(false, false, false) == "path-default");
    }

    public static void VerifyCompoundConditions()
    {
        Debug.Assert(PaymentRouter.Route("GET", true, !false) == "401-unauthorized");
        Debug.Assert(PaymentRouter.Route("GET", true, !true) == "200-read");
        Debug.Assert(PaymentRouter.EvaluateFlags(!false, !false, !true) == "path-alpha");
        Debug.Assert(PaymentRouter.EvaluateFlags(!true, !false, !false) == "path-beta");
    }

    public static void RunAllVerifications()
    {
        VerifyRouteRequestPaths();
        VerifyEvaluateFlagsCombinations();
        VerifyCompoundConditions();
    }
}
